File: Classes/Structure/Storage/storage.vb
''' <summary>
''' DataType representing any number of Section's as a Storage().
''' </summary>
Public Class storage
	Public Shared arraySection() As Section
	Private stringPathStorage As String
	Private integerSectionCounter As Integer
''' <summary>
''' Empty Sub.
''' </summary>
	Public Sub New()
		
	End Sub
''' <summary>
''' Get/Adjust the path of the Storage without Initialize() and Load().
''' </summary>
	Public Property StoragePath As String
		Get
			Return stringPathStorage
		End Get
		Set (Value As String)
			stringPathStorage = Value
		End Set
	End Property
''' <summary>
''' Resizes arraySection() and fill with New Section's.
''' </summary>
	Public Sub StorageInitialize()
		integerSectionCounter = 0
		ReDim arraySection(System.IO.Directory.GetDirectories(stringPathStorage, "*", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0))
		For Each folder As String In System.IO.Directory.GetDirectories(stringPathStorage, "*", System.IO.SearchOption.TopDirectoryOnly)
			arraySection(integerSectionCounter) = New Section
			arraySection(integerSectionCounter).SectionPath = folder
			integerSectionCounter += 1
		Next
	End Sub
''' <summary>
''' Loads all Shelves with all Bin's with all Items.
''' </summary>
	Public Shared Sub StorageLoad()
		For Each Section As section In arraySection
			Section.SectionInitialize()
			Section.SectionLoad()
		Next
	End Sub
''' <summary>
''' Adds a empty Storage with 1 Section and 1 Bin containing 'Nothing'.
''' </summary>
	Public Sub SectionAdd()
		Dim intCheck As Integer = 0
		Do
			If System.IO.Directory.Exists(stringPathStorage & "\" & intCheck) = False Then
				Exit Do
			End If
			intCheck += 1
		Loop
		ReDim Preserve arraySection(intCheck)
		System.IO.Directory.CreateDirectory(stringPathStorage & "\" & intCheck)
		arraySection(arraySection.GetUpperBound(0)) = New Section
		arraySection(arraySection.GetUpperBound(0)).SectionPath = stringPathStorage & "\" & intCheck
	End Sub
End Class

File: Game/Classes/Structure/Storage/storage.cs
using System;

/// <summary>
/// DataType representing any number of Section's as a Storage().
/// </summary>
public class storage
{
    public section[] arraySection;
    string stringPathStorage;
    int integerSectionCounter;

    /// <summary>
    /// Get/Adjust the path of the Storage without Initialize() and Load().
    /// </summary>
    public string StoragePath {
        get { return stringPathStorage; }
        set { stringPathStorage = value; }
    }

    /// <summary>
    /// Resizes arraySection() and fill with New Section's.
    /// </summary>
    public void StorageInitialize()
    {
        integerSectionCounter = 0;
        arraySection = new section[System.IO.Directory.GetDirectories(stringPathStorage, "*", System.IO.SearchOption.TopDirectoryOnly).GetUpperBound(0) + 1];
        foreach (string folder in System.IO.Directory.GetDirectories(stringPathStorage, "*", System.IO.SearchOption.TopDirectoryOnly)) {
            arraySection[integerSectionCounter] = new section();
            arraySection[integerSectionCounter].SectionPath = folder;
            integerSectionCounter += 1;
        }
    }
    
    /// <summary>
    /// Loads all Shelves with all Bin's with all Items.
    /// </summary>
    public void StorageLoad()
    {
        foreach (section Section in arraySection) {
            Section.SectionInitialize();
            Section.SectionLoad();
        }
    }

    /// <summary>
    /// Adds a empty Storage with 1 Section and 1 Bin containing 'Nothing'.
    /// </summary>
    public void SectionAdd()
    {
        int intCheck = 0;
        do {
            if (System.IO.Directory.Exists(stringPathStorage + "\\" + intCheck)) {
            } else {
                break; // TODO: might not be correct. Was : Exit Do
            }
            intCheck += 1;
        } while (true);
        Array.Resize(ref arraySection, intCheck + 1);
        System.IO.Directory.CreateDirectory(stringPathStorage + "\\" + intCheck);
        arraySection[arraySection.GetUpperBound(0)] = new section();
        arraySection[arraySection.GetUpperBound(0)].SectionPath = stringPathStorage + "\\" + intCheck;
    }
}